Dockerfile

```diff
@@ -0,0 +1,56 @@
+FROM debian:stable as builder
+
+RUN apt-get update && apt-get install -y --no-install-recommends curl git wget ca-certificates
+
+WORKDIR /java/
+
+# Download the JRE for copying to the image to run the OTP server
+RUN curl -Lo jre17.tar.gz https://github.com/adoptium/temurin17-binaries/releases/download/jdk-17.0.7%2B7/OpenJDK17U-jre_x64_linux_hotspot_17.0.7_7.tar.gz
+RUN tar xvf jre17.tar.gz && rm jre17.tar.gz
+
+# Download the JDK and maven and add them to path for building OTP and onebusaway-gtfs-modules
+RUN curl -Lo jdk17.tar.gz https://github.com/adoptium/temurin17-binaries/releases/download/jdk-17.0.7%2B7/OpenJDK17U-jdk_x64_linux_hotspot_17.0.7_7.tar.gz
+RUN tar xvf jdk17.tar.gz && rm jdk17.tar.gz
+ENV JAVA_HOME=/java/jdk-17.0.7+7/
+ENV PATH="$JAVA_HOME/bin:$PATH"
+
+RUN curl -Lo maven.tar.gz https://dlcdn.apache.org/maven/maven-3/3.9.1/binaries/apache-maven-3.9.1-bin.tar.gz
+RUN tar xvf maven.tar.gz && rm maven.tar.gz
+ENV PATH="/java/apache-maven-3.9.1/bin/:$PATH"
+
+# Build the MBTA branch of onebusaway-gtfs-modules so that it can be used in the OTP build
+WORKDIR /build/onebusaway-gtfs-modules
+# TODO: Remove the specific branch once the upgrade branch is merged into master
+RUN git clone --depth 1 -b otp2-upgrade --single-branch https://github.com/mbta/onebusaway-gtfs-modules .
+RUN ./build.sh
```

Comment:
Is there a more automated way to do it? I think it should be possible to include GitHub dependencies into pom.xml

Reply:
Hmm, I've never done it before, but it looks like there's a [third party site](https://jitpack.io/) that lets you use arbitrary github repos, at first I thought we'd need to pay for it, but it looks like any public open source repos are free, so we should be good to use it. The only other option seems to be publishing the onebusaway package to [github packages](https://docs.github.com/en/packages/working-with-a-github-packages-registry/working-with-the-apache-maven-registry) and using that as a mvn repository, which seems like a bit of a hassle. I'll try using jitpack to start and see how it goes.

Reply:
Ugh this is also sorta messy. In the [build script in OBA-gtfs-modules](https://github.com/mbta/onebusaway-gtfs-modules/blob/master/build.sh), we're skipping a bunch of build steps that are failing, like tests and a check for license headers in every file. This means that [jitpack fails to build the jar](https://jitpack.io/com/github/mbta/onebusaway-gtfs-modules/otp2-upgrade-onebusaway-gtfs-modules-1.3.63-gfcbe28e-576/build.log) and it can't be pulled by OTP. I put up https://github.com/mbta/onebusaway-gtfs-modules/pull/13 to get the clean install passing, trying to build locally using that branch with jitpack now.

Reply:
Still trying to work around this, after that the jitpack build succeeded and that's being resolved, but there are a couple other onebusaway dependencies that it uses, which were being installed to the local repository during the OBA install, but now are missing and causing an error when OTP tries to build.

Reply:
I'm still seeing issues with trying to use jitpack. Something is strange with the dependencies of the OBA-gtfs-modules repo. In its pom file, there are two other external OBA dependencies, with explicit versions set in the properties. When OTP is trying to use the version in the gtfs-modules pom file, which fails.

If I try to add those two dependencies explicitly to the OTP pom, it fails to build because a test fails with a missing apache commons CLI dependency, saying that a method added in version 1.3 doesn't exist, but when I do a `mvn dependency:tree`, it's not duplicated, with version 1.5 😐 Not totally sure how to proceed, but Github Packages are probably worth looking into instead (it looks like our OTP fork is already publishing packages there, so we can probably reuse a lot of that boilerplate from the actions), though they may very well lead to the same problem. Either way, I think the work for this might need to be broken out into a separate ticket, what do you think?

Reply:
Yeah, fair. Thanks for looking into it.

Reply:
Split out into [Use github packages for the onebusaway-gtfs-modules library](https://app.asana.com/0/1204355099788517/1204583577338745) and
[Update the OTP repo to fetch onebusaway dependency from Github Packages](https://app.asana.com/0/1204355099788517/1204583577338747).